### ISSUE TITLE
[Bug] Non-trivial Julia-like typing constructs from Hell raise unreadable exceptions... from Hell! :smiling_imp: 

### ISSUE THREAD

**Issue by ArneBachmann** [beartype/beartype#372]
With the latest release 0.18.4 in a pyodide browser WASM voodoo using micropip installation, I got

```python3
UnboundLocalError: cannot access local variable '__beartype_pith_2' where it is not associated with a value
```
and
```python3
UnboundLocalError: cannot access local variable '__beartype_pith_3 where it is not associated with a value
```
during a matplotlib action.
Doesn't happen with 0.17.2. nor with 0.18.2.

I was able to work around this by providing a fixed URL:
```js
await micropip.install(['https://files.pythonhosted.org/packages/ee/00/0ec51c742b402d353717663b5be56ac6aaf702932ad5b43de0ea7f996039/beartype-0.18.2-py3-none-any.whl', 'matplotlib']);

**Comment by leycec** [comment]
*...heh.* That's bad. Unfortunately, that's also *not* quite enough information to make this actionable. Do you have a minimal working example (MWE) of some sort that I can go off of? I'd love to help, because @beartype is clearly broken in some subtle way. But I need information to help. Tell me everything you know, @ArneBachmann! Everything! I'm even using the pleading face emoji. :pleading_face:

**Comment by ArneBachmann** [comment]
Reproducible outside pyodide, pure Linux Python venv 3.11.2:

```python3
File "/home/ash/Schreibtisch/all/projects/fiTrack/fitrack/misc.py", line 34, in fitrack.misc.transpose
Failed example:
    transpose([[1, 2, 3, 4], [5, 6, 7, 8], [9, 0, 1, 2]])
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.11/doctest.py", line 1351, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest fitrack.misc.transpose[1]>", line 1, in <module>
        transpose([[1, 2, 3, 4], [5, 6, 7, 8], [9, 0, 1, 2]])
      File "<@beartype(fitrack.misc.transpose) at 0x7eeea4b87ba0>", line 39, in transpose
    NameError: name '__beartype_pith_2' is not defined
```
happens here:
```python3
Number = int|float
NonNegativeNumber = Annotated[Number,Is[lambda n: n >= 0]]  # NonNegativeFloat | NonNegativeInt
NonEmptyStr = Annotated[str,Is[lambda s: s]]

TT = TypeVar('TT', bound=NonNegativeNumber|NonEmptyStr)

@beartype
def transpose(data:List[List[TT]]) -> List[List[TT]]:
  ''' Convert list with m lists of n values into n lists of m values each.

  Numbers only:
  >>> transpose([[1, 2, 3], [4, 5, 6]])
  [[1, 4], [2, 5], [3, 6]]
  >>> transpose([[1, 2, 3, 4], [5, 6, 7, 8], [9, 0, 1, 2]])
  [[1, 5, 9], [2, 6, 0], [3, 7, 1], [4, 8, 2]]

  With strings:
  >>> transpose([[1, 2, 3], [4, 5, 6], [';0', '2;2', '2|3;4|5']])
  [[1, 4, ';0'], [2, 5, '2;2'], [3, 6, '2|3;4|5']]
  '''
  cardinality:NonNegativeInt = len(data[0])
  return [[datum[i] for datum in data] for i in range(cardinality)]
```

I know, this code looks more like Julia than Python ;-)

**Comment by leycec** [comment]
**Woah.** Thanks so much for that concise expose of our life's failures, @ArneBachmann. When Python looks more like Julia than Python, you know that the `typing` community has finally achieved all its goals. :joy_cat: 

Of course, this issue is critically bad. @beartype's code generator has recently undergone... *changes.* Sometimes, changes good. Sometimes, changes bad. Recently, mostly bad. We broke a few omelettes to make an egg.

I'll *immediately* resolve this this weekend and excrete a new @beartype 0.18.5 patch release. The accursed @beartype 0.18.X release cycle continues to mock my desire for free time. Can't I just play video games all day like I used to? `</sigh>`

**Comment by leycec** [comment]
**Resolved locally.** Trust me, bro! Seriously, though. I'd like to run this through a gauntlet of razor-blade unit tests before releasing this as @beartype 0.18.5. But... this is looking good aside from being horribly broken. I might have a weekend after all! Everyone can now rest easy. :sweat_smile:

**Comment by leycec** [comment]
**Resolved by 7e9eef4.** So sorry for all the breakage in the 0.18.X series. Someday, this car without wheels will drive again. :smiling_face_with_tear: 

Until then, @beartype 0.18.5 is comin' right up. Someday, this will all be a distant memory we can laugh about.

**Comment by ArneBachmann** [comment]
That's awesome and super fast!
